### PR TITLE
api: GetPullRequestCommits: return file list

### LINKED ISSUE #27481: routers/api/v1/repo /pull.go: GetPullRequestCommits: empty Files list
### Description

The gitea API now returns an empty file list for commits:

https://try.gitea.io/api/v1/repos/msantos/test/pulls/1/commits

```json
[
  {
    "url": "https://try.gitea.io/api/v1/repos/msantos/test/git/commits/3a17cc9a6227c8c859851398ccc5a189f2bfdd6f",
    "sha": "3a17cc9a6227c8c859851398ccc5a189f2bfdd6f",
    "created": "2023-08-02T17:43:35-04:00",
    "html_url": "https://try.gitea.io/msantos/test/commit/3a17cc9a6227c8c859851398ccc5a189f2bfdd6f",
    "commit": ...,
    "author": ...,
    "committer": ...,
    "parents": [
      {
        "url": "https://try.gitea.io/api/v1/repos/msantos/test/git/commits/02f1e87c51f94fcaa3d3af904abd9ddc848b030e",
        "sha": "02f1e87c51f94fcaa3d3af904abd9ddc848b030e",
        "created": "0001-01-01T00:00:00Z"
      }
    ],
    "files": null,
    "stats": {
      "total": 2,
      "additions": 2,
      "deletions": 0
    }
  },
...
]
```

The file list is available using the files endpoint:

https://try.gitea.io/api/v1/repos/msantos/test/pulls/1/files

```json
[
  {
    "filename": "README.md",
    "status": "changed",
    "additions": 3,
    "deletions": 0,
    "changes": 3,
    "html_url": "https://try.gitea.io/msantos/test/src/commit/3a17cc9a6227c8c859851398ccc5a189f2bfdd6f/README.md",
    "contents_url": "https://try.gitea.io/api/v1/repos/msantos/test/contents/README.md?ref=3a17cc9a6227c8c859851398ccc5a189f2bfdd6f",
    "raw_url": "https://try.gitea.io/msantos/test/raw/commit/3a17cc9a6227c8c859851398ccc5a189f2bfdd6f/README.md"
  }
]
```

Untested patch:

```diff
diff --git a/routers/api/v1/repo/pull.go b/routers/api/v1/repo/pull.go
index 586f3385b..3bd2703e2 100644
--- a/routers/api/v1/repo/pull.go
+++ b/routers/api/v1/repo/pull.go
@@ -1330,7 +1330,7 @@ func GetPullRequestCommits(ctx *context.APIContext) {

        apiCommits := make([]*api.Commit, 0, end-start)
        for i := start; i < end; i++ {
-               apiCommit, err := convert.ToCommit(ctx, ctx.Repo.Repository, baseGitRepo, commits[i], userCache, convert.ToCommitOptions{Stat: true})
+               apiCommit, err := convert.ToCommit(ctx, ctx.Repo.Repository, baseGitRepo, commits[i], userCache, convert.ToCommitOptions{Stat: true, Files: true})
                if err != nil {
                        ctx.ServerError("toCommit", err)
                        return

```

### Gitea Version

1.20.5

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

* Github download
* systemd

### Database

SQLite

### PR DESCRIPTION
**Pull request by msantos**
Fixes https://github.com/go-gitea/gitea/issues/27481

---
Patch tested:

```json
[
  {
    "url": "http://100.115.92.198:9292/api/v1/repos/msantos/test/git/commits/7664dcb44167e0f9efd994e4ca6a9164694adc27",
    "sha": "7664dcb44167e0f9efd994e4ca6a9164694adc27",
    "created": "2023-10-06T09:57:08-04:00",
    "html_url": "http://100.115.92.198:9292/msantos/test/commit/7664dcb44167e0f9efd994e4ca6a9164694adc27",
...
    "files": [
      {
        "filename": "README.md",
        "status": "modified"
      }
    ],
    "stats": {
      "total": 2,
      "additions": 2,
      "deletions": 0
    }
  }
]
```

### CODE REVIEW

**Comment by silverwind** [comment]
Was this a regression? `files: null` does indicate i may have been.

**Comment by msantos** [comment]
> Was this a regression? `files: null` does indicate i may have been. If there are tests for this api call, I suggest validating presence of `files`.

Yes, I run some status checks using `files`. I'd have to check but I think it was working in 1.19. I'll look into the tests.

**Comment by silverwind** [comment]
Thanks, maybe you can also check the 1.20 and 1.21 branches so we know how far to backport this.

**Comment by msantos** [comment]
> Thanks, maybe you can also check the 1.20 and 1.21 branches so we know how far to backport this.

Introduced in https://github.com/go-gitea/gitea/commit/1dd83dbb917d55bd253001646d6743f247a4d98b released in 1.20.

I see the above commit made returning files/verification optional for the commit endpoint by using a parameter:

```
curl -X 'GET'   'https://try.gitea.io/api/v1/repos/msantos/test/commits?files=true'   -H 'accept: application/json'
````

The commit changed 3 endpoints: commit, pull and notes.

* pulls don't include verification
* notes don't include files or verification

Is it ok to have pulls always return files or should it be a paramter?

**Review by delvh — APPROVED** [review]

**Comment by msantos** [comment]
Updated to:
* return files/verification fields by default in the pulls and notes endpoints
* optionally disable using the `fields` and `verification` paramaters
* added tests

**Review by zeripath — APPROVED** [review]

File: routers/api/v1/repo/notes.go [hunk 1]
```diff
@@ -78,7 +78,15 @@ func getNote(ctx *context.APIContext, identifier string) {
 		return
 	}
 
-	cmt, err := convert.ToCommit(ctx, ctx.Repo.Repository, ctx.Repo.GitRepo, note.Commit, nil, convert.ToCommitOptions{Stat: true})
+	verification := ctx.FormString("verification") == "" || ctx.FormBool("verification")
+	files := ctx.FormString("files") == "" || ctx.FormBool("files")
```
Comment (wxiaoguang) [inline routers/api/v1/repo/notes.go:90]: Here and below, have these two new parameters been documented in swagger comment?

Comment (msantos) [reply routers/api/v1/repo/notes.go:90]: Thanks! Fixed.

**Comment by silverwind** [comment]
Given that previous `files` api was undocumented, I think we can strictly speaking not call it a regression.

**Comment by msantos** [comment]
> Given that previous `files` api was undocumented, I think we can strictly speaking not call it a regression, but I'd be willing to have it in 1.21.

https://github.com/go-gitea/gitea/commit/1dd83dbb917d55bd253001646d6743f247a4d98b added the new `files` api to the commits endpoint in 1.20 by modifying a function shared by the notes and pulls endpoints. The change unconditionally disabled returning values for the `files` and `verification` fields.

The change in this PR:
* notes/pulls: restores the behaviour in 1.19 of returning files/verification
* notes/pulls: allows optionally disabling files/verification by copying the `files` api from https://github.com/go-gitea/gitea/commit/1dd83dbb917d55bd253001646d6743f247a4d98b

The second item is unrelated to https://github.com/go-gitea/gitea/issues/27481 and could be dropped or split into a separate PR. Sorry for any confusion!

**Pull request merged**